File: app/assets/javascripts/Questionnaire.js
const mongoose = require('mongoose');
const { DataElementSchema } = require('./basetypes/DataElement');
const Code = require('./basetypes/Code');
const Interval = require('./basetypes/Interval');
const Quantity = require('./basetypes/Quantity');
const DateTime = require('./basetypes/DateTime');
const { ComponentSchema } = require('./Component');
const { FacilityLocationSchema } = require('./FacilityLocation');
const { IdSchema } = require('./Id');
const Any = require('./basetypes/Any');

const [Number, String] = [
  mongoose.Schema.Types.Number,
  mongoose.Schema.Types.String,
];

const QuestionnaireSchema = DataElementSchema({
  url: Any,
  identifier: Any,
  version: Any,
  name: Any,
  title: Any,
  derivedFrom: Any,
  status: Any,
  experimental: Any,
  subjectType: Any,
  date: Any,
  publisher: Any,
  contact: Any,
  description: Any,
  useContext: Any,
  jurisdiction: Any,
  purpose: Any,
  copyright: Any,
  approvalDate: Any,
  lastReviewDate: Any,
  effectivePeriod: Any,
  code: Any,
  item: Any,
  qdmVersion: { type: String, default: '4.0.0' },
  _type: { type: String, default: 'Questionnaire' },

});

module.exports.QuestionnaireSchema = QuestionnaireSchema;
module.exports.Questionnaire = mongoose.model('Questionnaire', QuestionnaireSchema);

File: app/models/qdm/questionnaire.rb
module QDM
  # app/models/qdm/questionnaire.rb
  class Questionnaire < DataElement
    include Mongoid::Document
    embedded_in :patient
    field :url
    field :identifier
    field :version
    field :name
    field :title
    field :derivedFrom
    field :status
    field :experimental
    field :subjectType
    field :date
    field :publisher
    field :contact
    field :description
    field :useContext
    field :jurisdiction
    field :purpose
    field :copyright
    field :approvalDate
    field :lastReviewDate
    field :effectivePeriod
    field :code
    field :item
    field :qdmVersion, type: String, default: '4.0.0'
  end
end
